File: LeetCode/Medium/0074-search-a-2d-matrix/0074-search-a-2d-matrix.c
bool searchMatrix(int** matrix, int matrixSize, int* matrixColSize, int target){
    int start = 0;
    int col = *matrixColSize;
    int end = matrixSize * col - 1;

    while(start <= end){
        int mid = (start + end) / 2;
        int value = matrix[mid/col][mid%col];
        if(value == target)
            return true;
        else if(value < target)
            start = mid + 1;
        else
            end = mid - 1;
    }
    return false;
}

File: LeetCode/Medium/0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        
        start, end = 0, m*n-1
        while start <= end:
            mid = (start + end)//2
            if matrix[mid//n][mid%n] == target:
                return True
            elif matrix[mid//n][mid%n] < target:
                start = mid + 1
            else:
                end = mid - 1
        return False    